### ISSUE TITLE
[A11y] Activating pagination bullets with keyboard doesn't work when Swiper is first initialized with size 0 

### ISSUE THREAD

**Issue by ReneZeidler** [nolimits4web/swiper#7817]
### Check that this is really a bug

- [X] I confirm

### Reproduction link

https://stackblitz.com/edit/swiper-pagination-a11y-bug

### Bug description

I encountered an issue where sometimes the pagination bullets are not activatable by keyboard even though `pagination.clickable` is set to true. This happens when the swiper element has size 0 during initialization, e.g. because one of its parent elements is hidden. In the reproduction I've simulated this by wrapping it inside a hidden div that gets unhidden after a delay, but in practical use I've also encountered this when the swiper isn't intentionally hidden and just has size 0 due to some initialization logic.

I backtraced the issue:

<details>

1. This is the code that adds the keyboard event listener to the pagination. It doesn't get run because `hasClickablePagination()` returns `false`. https://github.com/nolimits4web/swiper/blob/488636f0a7d3a8c6c26d11cf80c92fdff08c602e/src/modules/a11y/a11y.mjs#L333-L339
2. The method returns `false` because `swiper.pagination.bullets.length` is `0`. https://github.com/nolimits4web/swiper/blob/488636f0a7d3a8c6c26d11cf80c92fdff08c602e/src/modules/a11y/a11y.mjs#L175-L181
3. The `bullets` array is empty because `swiper.snapGrid` is empty.
https://github.com/nolimits4web/swiper/blob/488636f0a7d3a8c6c26d11cf80c92fdff08c602e/src/modules/pagination/pagination.mjs#L289
4. `snapGrid` is initialized to the empty array:
https://github.com/nolimits4web/swiper/blob/488636f0a7d3a8c6c26d11cf80c92fdff08c602e/src/core/core.mjs#L141
Usually, if it's empty, it gets set to the 1-element array `[0]` here:
https://github.com/nolimits4web/swiper/blob/488636f0a7d3a8c6c26d11cf80c92fdff08c602e/src/core/update/updateSlides.mjs#L234
However, that code doesn't get run because the function returns early when `swiper.size` is `undefined`:
https://github.com/nolimits4web/swiper/blob/488636f0a7d3a8c6c26d11cf80c92fdff08c602e/src/core/update/updateSlides.mjs#L43-L45
5. Finally, `swiper.size` only gets set when it has a size > 0:
https://github.com/nolimits4web/swiper/blob/488636f0a7d3a8c6c26d11cf80c92fdff08c602e/src/core/update/updateSize.mjs#L7-L20
When the swiper element is hidden during initialization, or has width 0 for some other reason, `size` doesn't get set.

</details>

I can see two potential fixes for this issue:

* It seems like the check for `bullets.length > 0` is unnecessary:  
```diff
 function hasPagination() { 
-   return swiper.pagination && swiper.pagination.bullets && swiper.pagination.bullets.length;
+   return swiper.pagination && swiper.pagination.bullets;
 } 
```
The only place where `bullets` is accessed is here, and `forEach` is safe to call on an empty array:
https://github.com/nolimits4web/swiper/blob/488636f0a7d3a8c6c26d11cf80c92fdff08c602e/src/modules/a11y/a11y.mjs#L186

* Since `snapGrid` gets set to `[0]` when it is empty in [`updateSlides.mjs`](https://github.com/nolimits4web/swiper/blob/488636f0a7d3a8c6c26d11cf80c92fdff08c602e/src/core/update/updateSlides.mjs#L234), it might be sensible to initialize it with that instead of an empty array:
https://github.com/nolimits4web/swiper/blob/488636f0a7d3a8c6c26d11cf80c92fdff08c602e/src/core/core.mjs#L141
```diff
-      snapGrid: [],
+      snapGrid: [0],
```
However, since I don't know the codebase, I don't know if that breaks any assumptions elsewhere.

Either (or both) of these changes should fix the issue by not skipping the initialization in the a11y module where the pagination keydown listener gets added.

### Expected Behavior

Focusing a clickable pagination bullet with the keyboard and then activating it `Enter` or `Space` should switch to the corresponding slide.

### Actual Behavior

Nothing happens.

### Swiper version

11.1.15

### Platform/Target and Browser Versions

Windows, Chrome 131/Firefox 131

### Validations

- [X] Follow our [Code of Conduct](https://github.com/nolimits4web/swiper/blob/master/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://swiperjs.com/swiper-api).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.
- [X] Make sure this is a Swiper issue and not a framework-specific issue

### Would you like to open a PR for this bug?

- [X] I'm willing to open a PR
